plugins/snowflake/dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -41,15 +52,15 @@
 
     {{ adapter.expand_target_column_types(from_relation=tmp_relation,
                                           to_relation=target_relation) }}
-    {% set incremental_sql %}
-    (
-        select * from {{ tmp_relation }}
-    )
-    {% endset %}
-
     {% set dest_columns = adapter.get_columns_in_relation(target_relation) %}
     {%- call statement('main') -%}
-      {{ get_merge_sql(target_relation, incremental_sql, unique_key, dest_columns) }}
+      {% if strategy == 'merge' %}
+        {{ get_merge_sql(target_relation, tmp_relation, unique_key, dest_columns) }}
+      {% elif strategy == 'delete+insert' %}
+        {{ get_delete_insert_merge_sql(target_relation, tmp_relation, unique_key, dest_columns) }}
+      {% else %}
+        {% do exceptions.raise_compiler_error('invalid strategy: ' ~ strategy) %}
```

Comment:
Shouldn't this just be a repeat of the above `invalid_strategy_msg`?

Reply:
yeah, i felt bad leaving the dangling `else` - this is like `InternalError: this should not be possible` territory haha

Reply:
I chose to hoist the error message to the top of the materialization so that users would get an error for a bad strategy on their first run (or a full-refresh) of an incremental model. That feels better to me than waiting for the second invocation to fail